0065-drop-dunder-utils.md

```diff
@@ -85,20 +101,48 @@ There are some things which we just can't control, see:
 ## Unresolved questions
 [unresolved]: #unresolved-questions
 
-None so far.
+### Many others may also be using ``salt://_utils/``. That can't be replaced with a simple import. That code would need to be packaged and installed separately into the minion's environment. — By [James Howe](https://github.com/OrangeDog)
+
+What exactly would be a use case for the ``salt://_utils/`` usage?
+
 
 ## Resolved questions
 [resolved]: #resolved-questions
 
-### Is this loader accessible anywhere else? e.g. via utils in templates? That significantly increases the surface of the breaking change. But that's not necessarily a bad thing.
+### Is this loader accessible anywhere else? e.g. via utils in templates? That significantly increases the surface of the breaking change. But that's not necessarily a bad thing. — By [James Howe](https://github.com/OrangeDog)
 
 Thank god no.
 
-### Breaking change on public API for custom modules (and extension modules?)
+### Breaking change on public API for custom modules (and extension modules?) — By [James Howe](https://github.com/OrangeDog)
 
 During the deprecation period, ``__utils__`` shall be available, although a warning about it's
 deprecation will be emmitted.
 
+### But still, is there a cleaner way to solve the original use-case (distribute common code, or override built-in utils)? — By [Max Arnold](https://github.com/max-arnold)
+
+Distributing common code has been solved in the python ecosystem, ie, packaged libraries.
+As for overriding built-in utils, this is one of the major reasons why the utils loader should have never been implemented.
+One thing is being able to override an execution module, or state module, the exposure surface is small, as for
+utility code, the surface is unmeasureable because that code is meant to be reused accross the whole code base. One small change
+in the overridden code has the potential of bringing a Salt installation to the ground.
+
+### I guess shipping custom utils through salt extensions will be also broken? — By [Max Arnold](https://github.com/max-arnold)
+
+A salt extension is a packaged python library that Salt can use, so, why not just import the utilitary code and just use it.
+
+### I do not mind to remove ``__utils__``, but what mechanism should I use now? — By [Alberto Planas](https://github.com/aplanas)
+
+Packaged libraries distributed to the minions. Salt can even distribute and install these on the minions.
```

Comment:
I can see some inconveniences with this approach:

1. `__utils__` is a kind of box for side functions, that do not have too much consistency to be packed via RPM or a wheel. Is supporting code, usually very ad-hoc, that we do no want to export with a public API (as in a salt module)
2. This makes the high state a two stage process if `__utils__` is used in the Jinja template (first a installation of the dependencies, and later apply the highstate)

It is also asymmetric: the `utils` in the core can be used as support and will be synchronized for use, but the user cannot extend them.

IMHO packaging them and expecting to be there before the highstate is not equivalent.

If the modules and states code can deliver submodules, then we will have an equivalent replacement.

Reply:
> Salt allows access to `__utils__` in a jinja template?!

Templates have access to execution modules, and they have access to `__utils__`.

Reply:
Someone asked on Slack how to call an utility function from Jinja, so I went ahead and implemented that as a custom module: https://gist.github.com/max-arnold/48b8cae9b4bb4d6d5f5479922184bc68

Reply:
> Templates have access to execution modules, and they have access to `__utils__`.

Thank god there's no direct access to ``__utils__`` from templates.


> Someone asked on Slack how to call an utility function from Jinja, so I went ahead and implemented that as a custom module: https://gist.github.com/max-arnold/48b8cae9b4bb4d6d5f5479922184bc68

For this they could register jinja filters, or when using your custom module, have it import the right utils and call it, there's really no need for ``__utils__``.